aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,228 @@
+      fileIO = new HadoopFileIO(conf);
+    }
+    return fileIO;
+  }
+
+  /**
+   * Returns table name in the form glue.catalogId.databaseName.tableName
+   * @return table name
+   */
+  @Override
+  protected String tableName() {
+    return fullName;
+  }
+
+  @Override
+  protected void doRefresh() {
+    String metadataLocation = null;
+    Optional<Table> tableOptional = getGlueTable();
+    if (tableOptional.isPresent()) {
+      Table table = tableOptional.get();
+      GlueToIcebergConverter.validateTable(table, tableName());
+      metadataLocation = table.parameters().get(METADATA_LOCATION_PROP);
+    } else {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException("Cannot find Glue table %s after refresh, " +
+            "maybe another process deleted it or revoked your access permission", tableName());
+      }
+    }
+    refreshFromMetadataLocation(metadataLocation);
+  }
+
+  @Override
+  protected void doCommit(TableMetadata base, TableMetadata metadata) {
+    String newMetadataLocation = writeNewMetadata(metadata, currentVersion() + 1);
+    boolean exceptionThrown = true;
+    boolean isUpdate = false;
+    Table glueTable = null;
+    try {
+      Optional<Table> glueTableOptional = getGlueTable();
+      if (glueTableOptional.isPresent()) {
+        glueTable = glueTableOptional.get();
+        // If we try to create the table but the metadata location is already set, then we had a concurrent commit
+        if (base == null && glueTable.parameters().get(METADATA_LOCATION_PROP) != null) {
+          throw new AlreadyExistsException("Cannot commit because table %s already exists in Glue", tableName());
+        }
+        isUpdate = true;
+        LOG.debug("Committing existing Glue table: {}", tableName());
+      } else {
+        LOG.debug("Committing new Glue table: {}", tableName());
+      }
+      checkMetadataLocation(isUpdate, glueTable, base);
+      Map<String, String> parameters = isUpdate ? Maps.newHashMap(glueTable.parameters()) : Maps.newHashMap();
+      updateParameters(parameters, newMetadataLocation);
+      persistGlueTable(isUpdate, parameters);
+      exceptionThrown = false;
+    } catch (CommitFailedException | AlreadyExistsException e) {
+      throw e;
+    } catch (ConcurrentModificationException e) {
+      throw new CommitFailedException(e,
+          "Cannot commit %s because Glue detected concurrent update", tableName());
+    } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
+      throw new AlreadyExistsException(e,
+          "Cannot commit %s because its Glue table already exists when trying to create one", tableName());
+    } catch (OperationTimeoutException | ResourceNumberLimitExceededException e) {
+      throw new CommitFailedException(e,
+          "Cannot commit %s because Glue operation or resource limit exceeded, " +
+              "please consider contact AWS to raise the limit", tableName());
```

Comment:
A timeout doesn't seem like the same case as exceeding a resource limit. Should these be handled separately?

Reply:
Yeah I was expecting OperationTimeoutException to catch throttling due to API rate limit exceeded, but it also catches other situations. it is probably better to make it a different case.